pandas-stubs/core/groupby/groupby.pyi

```diff
@@ -1,115 +1,436 @@
+        engine_kwargs: WindowingEngineKwargs = None,
+    ) -> NDFrameT: ...
+    @final
+    def max(
+        self,
+        numeric_only: bool = False,
+        min_count: int = -1,
+        engine: WindowingEngine = None,
+        engine_kwargs: WindowingEngineKwargs = None,
+    ) -> NDFrameT: ...
+    @final
+    def first(self, numeric_only: bool = False, min_count: int = -1) -> NDFrameT: ...
+    @final
+    def last(self, numeric_only: bool = False, min_count: int = -1) -> NDFrameT: ...
+    @final
     def ohlc(self) -> DataFrame: ...
-    def describe(self, **kwargs) -> DataFrame | Series: ...
-    def resample(self, rule, *args, **kwargs): ...
-    def rolling(self, *args, **kwargs): ...
-    def expanding(self, *args, **kwargs): ...
-    def ffill(self, limit: int | None = ...) -> DataFrame | Series: ...
-    def bfill(self, limit: int | None = ...) -> DataFrame | Series: ...
-    def nth(
-        self, n: int | list[int], dropna: str | None = ...
-    ) -> DataFrame | Series: ...
-    def quantile(self, q=..., interpolation: str = ...): ...
-    def ngroup(self, ascending: bool = ...) -> Series: ...
-    def cumcount(self, ascending: bool = ...) -> Series: ...
-    def rank(
+    def describe(
         self,
-        method: str = ...,
-        ascending: bool = ...,
-        na_option: str = ...,
-        pct: bool = ...,
-        axis: int = ...,
+        percentiles: Iterable[float] | None = None,
+        include: Literal["all"] | list[Dtype] | None = None,
+        exclude: list[Dtype] | None = None,
     ) -> DataFrame: ...
-    def cummax(self, axis: Axis = ..., **kwargs) -> DataFrame | Series: ...
-    def cummin(self, axis: Axis = ..., **kwargs) -> DataFrame | Series: ...
-    def cumprod(self, axis: Axis = ..., **kwargs) -> DataFrame | Series: ...
-    def cumsum(self, axis: Axis = ..., **kwargs) -> DataFrame | Series: ...
-    def shift(self, periods: int = ..., freq=..., axis: Axis = ..., fill_value=...): ...
+    @final
+    def resample(
+        self,
+        rule: Frequency,
+        # Arguments must be kept roughly inline with pandas.core.resample.get_resampler_for_grouping
+        how: str | None = None,
+        fill_method: str | None = None,
+        limit: int | None = None,
+        kind: str | None = None,
+        on: Hashable | None = None,
+        *,
+        closed: Literal["left", "right"] | None = None,
+        label: Literal["left", "right"] | None = None,
+        axis: Axis = 0,
+        convention: TimestampConvention | None = None,
+        origin: TimeGrouperOrigin | TimestampConvertibleTypes = "start_day",
+        offset: TimedeltaConvertibleTypes | None = None,
+        group_keys: bool = False,
+        **kwargs,
+    ) -> _ResamplerGroupBy[NDFrameT]: ...
+    @final
+    def rolling(
+        self,
+        # Arguments must be kept roughly inline with pandas.core.window.RollingGroupby
+        window: int | dt.timedelta | str | BaseOffset | BaseIndexer | None = None,
```

Comment:
I am afraid this will not work. There is an isinstance check at runtime with `dt.timedelta` (which also includes pandas Timedelta) that would fail if passed `np.timedelta64` for example (I tested and it does fail).